jupytext/quarto.py

```diff
@@ -0,0 +1,111 @@
+"""Jupyter notebook to Quarto Markdown and back, using 'quarto convert'"""
+
+import os
+import subprocess
+import tempfile
+
+# Copy nbformat reads and writes to avoid them being patched in the contents manager!!
+from nbformat import reads as ipynb_reads
+from nbformat import writes as ipynb_writes
+
+QUARTO_MIN_VERSION = "0.2.116"
+
+
+class QuartoError(OSError):
+    """An error related to Quarto"""
+
+
+def quarto(args, filein=None):
+    """Execute quarto with the given arguments"""
+    cmd = ["quarto"] + args.split()
+
+    if filein:
+        cmd.append(filein)
+
+    proc = subprocess.Popen(cmd, stdout=subprocess.PIPE)
+    out, err = proc.communicate()
+    if proc.returncode:
+        raise QuartoError(
+            f"{' '.join(cmd)} exited with return code {proc.returncode}\n{err}"
+        )
+    return out.decode("utf-8")
+
+
+def is_quarto_available(min_version=QUARTO_MIN_VERSION):
+    """Is Quarto available?"""
+    try:
+        raise_if_quarto_is_not_available(min_version=min_version)
+        return True
+    except QuartoError:
+        return False
+
+
+def raise_if_quarto_is_not_available(min_version=QUARTO_MIN_VERSION):
+    """Raise with an informative error message if quarto is not available"""
+    try:
+        from pkg_resources import parse_version
+    except ImportError:
+        raise QuartoError("Please install pkg_resources")
+
+    version = quarto_version()
+    if version == "N/A":
+        raise QuartoError(
+            f"The Quarto Markdown format requires 'quarto>={min_version}', "
+            "but quarto was not found"
+        )
+
+    if parse_version(version) < parse_version(min_version):
+        raise QuartoError(
+            f"The Quarto Markdown format requires 'quarto>={min_version}', "
+            f"but quarto version {version} was not found"
+        )
+
+    return version
+
+
+def quarto_version():
+    """Quarto's version number"""
+    try:
+        return quarto("--version").strip()
```

Comment:
As long as there is negligible overhead for `quarto` invocations I don't think that this check is worth the execution time. Quarto is early enough in its dev cycle that we can reasonably expect that nearly all users will have a recent version (and we can reference the minimum required version in the jupytext docs). 

If this isn't persuasive to you then I'd suggest two other approaches to avoid the overhead:

1) Only detect the `quarto_version` after an error has occurred (and in that case print the informative message that they need a newer version; or 
2) Detect the quarto version from the filesystem. Once you know the directory containing `quarto` you can find the version file at `../share/version` (if there is no version file then it's the development version which you can assume meets the minimum criteria).

Reply:
I have one idea about how to make `quarto --version` faster so that no filesystem check is required --- will report back on whether this worked out soon.